qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,312 @@
+            netperf_test_duration = int(
+                params.get("netperf_test_duration", 180))
+            netperf_para_sess = params.get("netperf_para_sessions", "1")
+            test_protocols = params.get("test_protocols", "TCP_STREAM")
+            netperf_cmd_prefix = params.get("netperf_cmd_prefix", "")
+            netperf_output_unit = params.get("netperf_output_unit")
+            netperf_package_sizes = params.get("netperf_sizes")
+            test_option = params.get("test_option", "")
+            test_option += " -l %s" % netperf_test_duration
+            if params.get("netperf_remote_cpu") == "yes":
+                test_option += " -C"
+            if params.get("netperf_local_cpu") == "yes":
+                test_option += " -c"
+            if netperf_output_unit in "GMKgmk":
+                test_option += " -f %s" % netperf_output_unit
+            start_time = time.time()
+            stop_time = start_time + netperf_test_duration
+            for protocol in test_protocols.split():
+                t_option = "%s -t %s" % (test_option, protocol)
+                n_client.bg_start(utils_net.get_host_ip_address(params),
+                                  t_option,
+                                  netperf_para_sess, netperf_cmd_prefix,
+                                  package_sizes=netperf_package_sizes)
+                if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+                                       "Wait netperf test start"):
+                    logging.info("Netperf test start successfully.")
+                else:
+                    test.error("Can not start netperf client.")
+                # here when set a run flag, when other case call this case as a
+                # subprocess backgroundly,can set this run flag to False to stop
+                # the stress test.
+                env["netperf_run"] = True
+                netperf_test_duration = stop_time - time.time()
+                utils_misc.wait_for(
+                    lambda: not n_client.is_netperf_running(),
+                    netperf_test_duration, 0, 5,
+                    "Wait netperf test finish %ss" % netperf_test_duration)
+                time.sleep(5)
+        finally:
+            n_server.stop()
+            n_server.package.env_cleanup(True)
+            n_client.package.env_cleanup(True)
+
+    login_timeout = int(params.get("login_timeout", 360))
+    bg_stress_test = params.get("run_bgstress")
+    bg_stress_run_flag = params.get("bg_stress_run_flag")
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login(timeout=login_timeout)
+    enable_multi_queues(vm)
+    session_serial = vm.wait_for_serial_login(timeout=login_timeout)
+    b_ping_lost_ratio = int(params.get("background_ping_package_lost_ratio", 5))
+    b_ping_time = int(params.get("background_ping_time", 60))
+    guest_ip = vm.get_address()
+    bg_test = None
+
+    try:
+
```

Comment:
Not need blank line.

Reply:
I have solved the above problems, please help me review it again,thanks a lot.